library/core/src/fmt/mod.rs

```diff
@@ -340,7 +340,9 @@ impl<'a> Arguments<'a> {
     #[rustc_const_unstable(feature = "const_fmt_arguments_new", issue = "none")]
     pub const fn new_const(pieces: &'a [&'static str]) -> Self {
```

Comment:
Is there a reason it can't be 

```suggestion
    pub const fn new_const(pieces: &'a [&'static str; 1]) -> Self {
```
eliminating the "runtime" check?

Reply:
This function currently is called with references to arrays of length 1 and 0. This is a decent idea, but at least it would require some amount of surgery on the `format_args!` expansion code.

Reply:
What about then?
```suggestion
    pub const fn new_const<const N: usize>(pieces: &'a [&'static str; N]) -> Self {
```
After monomorphization the check should be eliminat-able this way, though I suspect this function is probably small enough to be probably always inlined anyways, making this point moot.